src/backend/joanie/tests/test_api_address.py

```diff
@@ -0,0 +1,325 @@
+            f"/api/addresses/{address.pk}/",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+            data=bad_data,
+        )
+        self.assertEqual(response.status_code, 400)
+        content = json.loads(response.content)
+        self.assertEqual(content, {"errors": {"name": ["This field is required."]}})
+
+    def test_create_update_address(self):
+        """Create/update user addresses with valid token"""
+        username = "panoramix"
+        token = self._mock_user_token(username)
+        data = {
+            "name": "Home",
+            "address": "61 bis rue de la Glacière",
+            "postcode": "75013",
+            "city": "Paris",
+            "country": "FR",
+        }
+
+        response = self.client.post(
+            "/api/addresses/",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+            data=data,
+        )
+        self.assertEqual(response.status_code, 201)
+
+        # panoramix was a unknown user, so a new user was created
+        owner = models.User.objects.get()
+        self.assertEqual(owner.username, username)
+
+        # new address was created for user panoramix
+        address = models.Address.objects.get()
+        self.assertEqual(address.owner, owner)
+        self.assertEqual(address.city, "Paris")
+
+        data["name"] = "Office"
+        response = self.client.post(
+            f"/api/addresses/{address.pk}/",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+            data=data,
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(models.Address.objects.count(), 1)
+        address = models.Address.objects.get()
+        self.assertEqual(address.name, "Office")
+        self.assertEqual(address.owner, owner)
+        self.assertEqual(address.city, "Paris")
+
+    def test_delete_without_authorization(self):
+        """Delete address is not allowed without authorization"""
+        username = "panoramix"
+        user = factories.UserFactory(username=username)
+        address = factories.AddressFactory.create(owner=user, name="Office")
+        response = self.client.delete(
+            f"/api/addresses/{address.pk}/",
+        )
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_with_bad_authorization(self):
+        """Delete address is not allowed with bad authorization"""
+        user = factories.UserFactory(username="panoramix")
+        address = factories.AddressFactory.create(owner=user)
+        response = self.client.delete(
+            f"/api/addresses/{address.pk}/",
+            HTTP_AUTHORIZATION="Bearer nawak",
+        )
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_with_expired_token(self):
+        """Delete address is not allowed with expired token"""
+        token = self._mock_user_token(
+            "panoramix",
+            expires_at=arrow.utcnow().shift(days=-1).datetime,
+        )
+        user = factories.UserFactory(username="panoramix")
```

Comment:
you could let factory boy generate the username and get the token after generating the user:
```suggestion
        user = factories.UserFactory()
        token = self._mock_user_token(
            user.username,
            expires_at=arrow.utcnow().shift(days=-1).datetime,
        )
```